lib/proxy/sd_notify.rb

```diff
@@ -0,0 +1,27 @@
+require 'socket'
+
+# Implementation of libsystemd's sd_notify API, sends current state via socket
+module Proxy
+  class SdNotify
+    def active?
+      !ENV['NOTIFY_SOCKET'].nil?
+    end
+
+    def notify(message)
+      socket.sendmsg(message.chomp + "\n") # ensure trailing \n
```

Comment:
It was relying on the GC to finalise the object, which would autoclose it. I've changed it to explicitly close the socket as soon as it's done to free up the fd.